### PR TITLE
add: python finetuning scripts

### PR DESCRIPTION
**Pull request by laurafcamargos**
This PR introduces fine-tuning scripts for a Question Answering model using Hugging Face Transformers. 

* feat: update comments
* fix: update functions
* fix: update variables

### CODE REVIEW

File: python/document_reader_fine_tuning.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+Original file is located at
+    https://colab.research.google.com/drive/1PRCbaNy2B4vs_3oO2PaOSswJPZUL0PTD
+"""
+
+# pip install datasets
+# pip install transformers
+# pip install huggingface_hub
+# pip install evaluate
+# pip install torch
```
Comment (leomaurodesenv) [inline python/document_reader_fine_tuning.py]: You can remove these comments

File: python/document_reader_fine_tuning.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    TrainingArguments,
+    Trainer,
+    EarlyStoppingCallback,
+    pipeline,
+)
+from huggingface_hub import HfApi, login
+
+DOC_READER = DocReader.DistilBERT
+SPORT = Sports.BASKETBALL
+
```
Comment (leomaurodesenv) [inline python/document_reader_fine_tuning.py]: Please, add a comment explaining the propose of these variables.
After the selection, add a `print(SPORT.name, DOC_READER.name)`

File: python/document_reader_fine_tuning.py [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+os.environ["HUGGINGFACE_TOKEN"] = HUGGINGFACE_TOKEN
+login(token=HUGGINGFACE_TOKEN)
+
+
+# Carregar dataset
```
Comment (leomaurodesenv) [inline python/document_reader_fine_tuning.py]: Change the comments to English

File: python/utils.py [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+
+def filter_dataset(dataset: Dataset) -> Dataset:
+    return dataset.filter(
+        lambda x: x["start_positions"] != 0 #remove todos os exemplos onde {'text': '', 'offset': [0, 0]}
+        and x["end_positions"] != 0
+        and x["start_positions"] <= x["end_positions"]  #garante que o início <= fim
+    )
+
+def dataset_load(split: str):
+    load_dataset("PedroCJardim/QASports", SPORT, split = split)
```
Comment (leomaurodesenv) [inline python/utils.py]: Change SPORT to be a parameter. This variable does not exist in this file

File: python/document_reader_fine_tuning.py (hunk 3 above)
Comment (leomaurodesenv) [reply python/document_reader_fine_tuning.py]: Review all document, comments, etc

File: python/document_reader_fine_tuning.py [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+    eval_dataset=filtered_data_validation,  # Dataset de validação processado
+    data_collator=data_collator,  # Collator correto
+    compute_metrics=compute_metrics,
+    callbacks=[EarlyStoppingCallback(early_stopping_patience=5)],
+)
+
+trainer.train()
+
+# Salvar tokenizer localmente
+# tokenizer.save_pretrained("distilbert-qasports")
```
Comment (leomaurodesenv) [inline python/document_reader_fine_tuning.py]: You can save using the name `DOC_READER.name` for simplification; something file `f"{str(DOC_READER.name).lower()}-{str(SPORT.name).lower()}"`

File: python/document_reader_fine_tuning.py [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+
+# Enviar tokenizer para o Hugging Face
+from huggingface_hub import HfApi
+
+api = HfApi()
+api.upload_folder(
+    folder_path="distilbert-qasports",
+    repo_id="laurafcamargos/distilbert-qasports",
+    commit_message="Adicionando tokenizer",
```
Comment (leomaurodesenv) [inline python/document_reader_fine_tuning.py]: The same, compose the name based on the doc_reader and sport variables

File: python/requirements.txt [hunk 7]
```diff
@@ -0,0 +1,8 @@
+# dev
+black==25.1.0
+
+# fine tuning
+datasets==3.3.1
+transformers==4.49.0
+huggingfhuggingface-hub==0.29.0
+torch==
```
Comment (leomaurodesenv) [inline python/requirements.txt]: it is missing the `torch` version

File: python/utils.py [hunk 8]
```diff
@@ -0,0 +1,159 @@
...
+
+def compute_metrics(p):
+    metric = load("squad_v2")
```
Comment (leomaurodesenv) [inline python/utils.py:137]: Why do you compute metrics using Squad and no the QASports?

File: python/document_reader_fine_tuning.py [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+    callbacks=[EarlyStoppingCallback(early_stopping_patience=5)],
+)
+
+trainer.train()
+
+# Salvar tokenizer localmente
+# tokenizer.save_pretrained("distilbert-qasports")
+
+# Enviar tokenizer para o Hugging Face
+from huggingface_hub import HfApi
```
Comment (leomaurodesenv) [inline python/document_reader_fine_tuning.py]: Move all imports to the begining to the document. Additionaly, run `black python/.` to fix all the issues in the document

**Review by leomaurodesenv — CHANGES_REQUESTED** [review]

**Comment by leomaurodesenv** [comment]
> _ No description provided. _

Please, add a proper description saying the intention and proposal of the Pull Request

File: python/utils.py (hunk 8 above)
Comment (laurafcamargos) [reply python/utils.py:137]: The squad_v2 is used because it provides a established evaluation metric for question-answering tasks, which includes F1-score and Exact Match and another 11 values. Since QASports is a dataset, not an evaluation metric, we need a predefined metric like squad_v2 to evaluate the model’s performance. Ref: https://huggingface.co/spaces/evaluate-metric/squad_v2

**Review by leomaurodesenv — APPROVED** [review]
LGTM
